Documentation/special-services/pod-mutator.md

```diff
@@ -0,0 +1,76 @@
+# PodMutator Service
+To provide pod mutation capabilities for finer-grained control over rkt containerization concepts and debugging needs, 
+this proposal introduces a new optional stage1 app/service with special semantics that is lightweight.
```

Comment:
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  gtrevg@gmail.com
    Domain plentyproductions.com has exceeded the max emails per hour (200/200 (100%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext3.iad.github.net ([192.30.252.194]:39765 helo=github-smtp2b-ext-cp1-prd.iad.github.net)
	by server21.abstractdns.com with esmtps (TLSv1:DHE-RSA-AES256-SHA:256)
	(Exim 4.87)
	(envelope-from <noreply@github.com>)
	id 1bLC4g-0001XP-Pn
	for github@trevorgattis.com; Thu, 07 Jul 2016 16:22:50 +0000
Date: Tue, 05 Jul 2016 12:55:28 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1467748528;
	bh=cKJB/NCupvRH1Rv9EdcQHSBpym7HnmH2+06ilU0bCIc=;
	h=From:Reply-To:To:Cc:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=2HBUDfHB26UpzXehnRwsIdI6FTGtSo5DdwxCQEHQ8NxJn5QfRP49WrAa5/o+yoeVP
	 CxcwEFUQOAH/Qp0N7W9sHiit4OOoR3zX1giiAQZ6krS3krkqFDcb5EWIMp4EJ4jaf3
	 3jSYrrMV3bxvJXuhpLRdgL/1SLJC7XbaqPWyqMh0=
From: Euan Kemp <notifications@github.com>
Reply-To: coreos/rkt <reply@reply.github.com>
To: coreos/rkt <rkt@noreply.github.com>
Cc:
Message-ID: <coreos/rkt/pull/2867/r69626737@github.com>
In-Reply-To: <coreos/rkt/pull/2867@github.com>
References: <coreos/rkt/pull/2867@github.com>
Subject: Re: [coreos/rkt] stage1: add pod mutability as a service (#2867)
Mime-Version: 1.0
Content-Type: multipart/alternative;
 boundary="--==_mimepart_577c10b03adac_5bd23f8119f572b85423d";
 charset=UTF-8
Content-Transfer-Encoding: 7bit
Precedence: list
X-GitHub-Sender: euank
X-GitHub-Recipient: gtrevg
List-ID: coreos/rkt <rkt.coreos.github.com>
List-Archive: https://github.com/coreos/rkt
List-Post: <mailto:reply@reply.github.com>
List-Unsubscribe: <mailto:unsub+001038eda04c850f5e342963f6fedc5ab56db9b0d6433fb892cf000000011393d2b092a170ce04266b71@reply.github.com>,
 <https://github.com/notifications/unsubscribe/ABA47ctKjtCq6FYjzz6tBXtTooagk0mQks5qSrawgaJpZM4JEHLz>
X-Auto-Response-Suppress: All
X-GitHub-Recipient-Address: github@trevorgattis.com


----==_mimepart_577c10b03adac_5bd23f8119f572b85423d
Content-Type: text/plain;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

> @@ -0,0 +1,76 @@
> +# PodMutator Service
> +To provide pod mutation capabilities for finer-grained control over rkt containerization concepts and debugging needs, 
> +this proposal introduces a new optional stage1 app/service with special semantics that is lightweight.

It kinda depends how first-class you want to make container support imo.

If you have a similar scheme on a per-app basis as we currently have on a per-pod basis (locks and states), or so on then you can use attempts to acquire that lock as a means to get status events.

Proactive events vs polling status is not a key distinction in the CRI and, while events are preferred, they're not explicitly required. Hooks (pre/post start/stop), iiuc, are going to be implemented externally in kubelet so they're not related to this.

---
You are receiving this because you are subscribed to this thread.
Reply to this email directly or view it on GitHub:
https://github.com/coreos/rkt/pull/2867/files/a2a943cc341e58a9dee09bb122f2e22f038e18e0#r69626737
----==_mimepart_577c10b03adac_5bd23f8119f572b85423d
Content-Type: text/html;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

<p>In <a href="https://github.com/coreos/rkt/pull/2867#discussion_r69626737">Documentation/special-services/pod-mutator.md</a>:</p>
<pre style='color:#555'>&gt; @@ -0,0 +1,76 @@
&gt; +# PodMutator Service
&gt; +To provide pod mutation capabilities for finer-grained control over rkt containerization concepts and debugging needs, 
&gt; +this proposal introduces a new optional stage1 app/service with special semantics that is lightweight.
</pre>
<p>It kinda depends how first-class you want to make container support imo.</p>

<p>If you have a similar scheme on a per-app basis as we currently have on a per-pod basis (locks and states), or so on then you can use attempts to acquire that lock as a means to get status events.</p>

<p>Proactive events vs polling status is not a key distinction in the CRI and, while events are preferred, they're not explicitly required. Hooks (pre/post start/stop), iiuc, are going to be implemented externally in kubelet so they're not related to this.</p>

<p style="font-size:small;-webkit-text-size-adjust:none;color:#666;">&mdash;<br />You are receiving this because you are subscribed to this thread.<br />Reply to this email directly, <a href="https://github.com/coreos/rkt/pull/2867/files/a2a943cc341e58a9dee09bb122f2e22f038e18e0#r69626737">view it on GitHub</a>, or <a href="https://github.com/notifications/unsubscribe/ABA47ctKjtCq6FYjzz6tBXtTooagk0mQks5qSrawgaJpZM4JEHLz">mute the thread</a>.<img alt="" height="1" src="https://github.com/notifications/beacon/ABA47VumiXveeiBzOnULvnD0J2FIeVNZks5qSrawgaJpZM4JEHLz.gif" width="1" /></p>
<div itemscope itemtype="http://schema.org/EmailMessage">
<div itemprop="action" itemscope itemtype="http://schema.org/ViewAction">
  <link itemprop="url" href="https://github.com/coreos/rkt/pull/2867/files/a2a943cc341e58a9dee09bb122f2e22f038e18e0#r69626737"></link>
  <meta itemprop="name" content="View Pull Request"></meta>
</div>
<meta itemprop="description" content="View this Pull Request on GitHub"></meta>
</div>

----==_mimepart_577c10b03adac_5bd23f8119f572b85423d--